src/NClient.Tools/NClient.DotNetTool/Options/CommonGenerationOptions.cs

```diff
@@ -0,0 +1,53 @@
+
+namespace NClient.DotNetTool.Options
+{
+    public abstract class CommonGenerationOptions : CommonOptions
+    {
+        [Option(shortName: 'a', longName: "api", Required = true, HelpText = "The OpenAPI spec file to use. Paths are relative to the project directory. You can also set a http URI as path")]
+        public virtual string Spec { get; }
+
+        [Option(shortName: 'o', longName: "output", Required = false, HelpText = "The result. File paths are relative to the project directory.", Default = "./Facades.cs")]
+        public virtual string OutputPath { get; }
+
+        [Option(shortName: 's', longName: "namespace", Required = false, HelpText = "The namespace for generated files.", Default = "NClient.Facades")]
+        public virtual string Namespace { get; }
+
+        [Option(longName: "withModelValidation", Required = false, HelpText = "The flag indicating whether to add model validation attributes.", Default = false)]
+        public virtual bool UseModelValidationAttributes { get; }
+            
+        [Option(longName: "withDto", Required = false, HelpText = "The flag indicating whether to generate DTO classes.", Default = true)]
+        public virtual bool GenerateDtoTypes { get; }
+
+        [Option(longName: "withCancellationTokens", Required = false, HelpText = "The flag indicating whether to allow adding cancellation token.", Default = false)]
+        public virtual bool UseCancellationToken { get; }
+
+        [Option(longName: "useSystemTextJson", SetName = "serializer", Required = false, HelpText = "The flag indicating the use of the SystemTextJson library for serialization.", Default = false)]
+        public virtual bool UseSystemTextJson { get; }
+            
+        [Option(longName: "useNewtonsoftJson", SetName = "serializer", Required = false, HelpText = "The flag indicating the use of the NewtonsoftJson library for serialization.", Default = false)]
+        public virtual bool UseNewtonsoftJson { get; }
+            
+        [Option(longName: "nullableEnable", Required = false, HelpText = "The flag indicating whether to generate Nullable Reference Type annotations.", Default = false)]
+        public virtual bool UseNullableReferenceTypes { get; }
+        
+        protected CommonGenerationOptions(
+            string spec, string outputPath, string @namespace,
+            bool useModelValidationAttributes, bool useNullableReferenceTypes,
+            bool generateDtoTypes, bool useCancellationToken,
+            bool useSystemTextJson, bool useNewtonsoftJson,
+            LogLevel logLevel) : base(logLevel)
+        {
+            Spec = spec;
+            OutputPath = outputPath;
+            Namespace = @namespace;
+            UseModelValidationAttributes = useModelValidationAttributes;
+            UseNullableReferenceTypes = useNullableReferenceTypes;
+            GenerateDtoTypes = generateDtoTypes;
+            UseCancellationToken = useCancellationToken;
```

Comment:
Maybe cancellation and dto is fundamental options without configuration? I don't see any relevant example when we need generate interfaces without dto and async methods without cancellation token.

Reply:
I partially agree, tokens can always be generated for clients, but this may be redundant for facades, since they are used to create controllers. To make it consistent, I added `UseCancellationToken` parameter both there and there.
Now about the dto. Since the client and the facade can be generated separately, double generation of the dto will lead to errors. Therefore, disabling this feature is still necessary

Reply:
Firstly... What's problem to use CancellationToken in controllers? I've missed this. 
Secondly: I think we shot ourselves in the leg with current arch design. Think about it.  First we'd been creating the tool for simple generation boilerplate code by OpenApi spec, then we've added additional functions with a separate facades and clients. And now we have a problem with double DTOs. I'm strongly convinced that totally wrong solution. We should re-design arch of generation process or simplify that.

Reply:
> Firstly... What's problem to use CancellationToken in controllers? I've missed this.

Judging by my practice, controllers with tokens are rare. If tokens will be generated for controllers, but will not be used, then IDEs may generate warnings because of unused parameters.

> Secondly: I think we shot ourselves in the leg with current arch design. Think about it. First we'd been creating the tool for simple generation boilerplate code by OpenApi spec, then we've added additional functions with a separate facades and clients. And now we have a problem with double DTOs. I'm strongly convinced that totally wrong solution. We should re-design arch of generation process or simplify that.

In any case, the functionality of this tool will expand and become more complex over time, I'm sure of it.
The added methods don't seem complicated to me, because all the parameters except one (--api) are optional. Moreover, you can generate everything at once with one simple command:
```
dotnet nclient facade generate --api http://localhost:8080/swagger.json --withClients
```

Reply:
> Judging by my practice, controllers with tokens are rare. If tokens will be generated for controllers, but will not be used, then IDEs may generate warnings because of unused parameters.

I'm sorry.. what? Simple example: unstable internet connection. Our request to web service could be interrupt in many cases.  The one suite, what I can image, it's some service without any deps like db. What are the odds? The fact that this is a rare solution does not mean that it is a correct solution.

> In any case, the functionality of this tool will expand and become more complex over time,

Ok, I agree. However, the fact that we increase cyclomatic complexity "by design" does not make me happy. If we made so one day we'll achieve much more entangled source code with no option to support.

Reply:
> I'm sorry.. what? Simple example: unstable internet connection. Our request to web service could be interrupt in many cases. The one suite, what I can image, it's some service without any deps like db. What are the odds? The fact that this is a rare solution does not mean that it is a correct solution.

I don't understand how the cancellation token will solve the problem of internet connection.

> Ok, I agree. However, the fact that we increase cyclomatic complexity "by design" does not make me happy. If we made so one day we'll achieve much more entangled source code with no option to support.

We will get entangled code in the future if we do not provide for the possibility of extension in advance :)

Reply:
The cancellation token in controllers is most often used if the method executes some kind of long operation. It is impractical to use it always, because it complicates the code and can create problems related to idempotency and atomicity. Moreover, if you need a method that takes a long time to execute, then perhaps you should think about asynchronous communication with the service. This is why cancellation tokens are rarely used in controllers.

Reply:
this is not convincing, but ok, I accept this.